core-tests/shared/src/test/scala/zio/StackTracesSpec.scala

```diff
@@ -84,54 +96,229 @@ object StackTracesSpec extends ZIOBaseSpec {
         for {
           _     <- ZIO.succeed(25)
           value  = underlyingFailure
-          trace <- matchPrettyPrintCause(value)
-        } yield {
-          assertHasExceptionInThreadZioFiber(trace)("java.lang.String: Oh no!") &&
-          assertHasStacktraceFor(trace)("spec.underlyingFailure") &&
-          assertHasStacktraceFor(trace)("matchPrettyPrintCause") &&
-          assert(trace)(containsString("Suppressed: java.lang.RuntimeException: other failure")) &&
-          assertTrue(numberOfOccurrences("Suppressed")(trace) == 1)
-        }
+          cause <- value.cause
+        } yield assert(cause)(causeHasTrace {
+          """java.lang.String: Oh no!
+            |	at zio.StackTracesSpec.spec.underlyingFailure
+            |	at zio.StackTracesSpec.spec
+            |	Suppressed: java.lang.RuntimeException: other failure
+            |		at zio.StackTracesSpec.spec.underlyingFailure
+            |		at zio.StackTracesSpec.spec
+            |""".stripMargin
+        })
       },
       test("captures a die failure") {
         val underlyingFailure =
           ZIO
             .succeed("ITEM")
             .map(_ => List.empty.head)
 
-        for {
-          trace <- matchPrettyPrintCause(underlyingFailure)
-        } yield {
-          assertHasExceptionInThreadZioFiber(trace)("java.util.NoSuchElementException: head of empty list") &&
-          assertHasStacktraceFor(trace)("spec.underlyingFailure") &&
-          assertHasStacktraceFor(trace)("matchPrettyPrintCause")
+        for (cause <- underlyingFailure.cause)
+          yield assert(cause)(causeHasTrace {
+            if (TestVersion.isScala2)
+              """java.util.NoSuchElementException: head of empty list
+                |	at scala.collection.immutable.Nil$.head
+                |	at zio.StackTracesSpec$.$anonfun$spec
+                |	at zio.ZIO.$anonfun$map
+                |	at zio.StackTracesSpec.spec.underlyingFailure
+                |	at zio.StackTracesSpec.spec
+                |""".stripMargin
+            else
+              """java.util.NoSuchElementException: head of empty list
+                |	at scala.collection.immutable.Nil$.head
+                |	at zio.StackTracesSpec$.$anonfun
+                |	at zio.ZIO.map$$anonfun
+                |	at zio.StackTracesSpec.spec.underlyingFailure
+                |	at zio.StackTracesSpec.spec
+                |""".stripMargin
```

Comment:
These 2 stack traces look the same to me, or did I miss some difference between them?

Reply:
`zio.ZIO.$anonfun$map` vs `zio.ZIO.map$$anonfun` 😄